File: src/utils/worker.c
/* Stellarium Web Engine - Copyright (c) 2018 - Noctua Software Ltd
 *
 * This program is licensed under the terms of the GNU AGPL v3, or
 * alternatively under a commercial licence.
 *
 * The terms of the AGPL v3 license can be found in the main directory of this
 * repository.
 */

#include "worker.h"
#include <string.h>

#ifndef HAVE_PTHREAD

void worker_init(worker_t *w, int (*fn)(worker_t *w))
{
    memset(w, 0, sizeof(*w));
    w->fn = fn;
}

int worker_iter(worker_t *w)
{
    if (w->state) return 1;
    w->fn(w);
    w->state = 1;
    return 1;
}

bool worker_is_running(worker_t *w)
{
    return false;
}

#endif

File: web-frontend/src/plugins/smt/query-engine/worker.js
// Stellarium Web - Copyright (c) 2020 - Stellarium Labs SAS
//
// This program is licensed under the terms of the GNU AGPL v3, or
// alternatively under a commercial licence.
//
// The terms of the AGPL v3 license can be found in the main directory of this
// repository.
//
// This file is part of the Survey Monitoring Tool plugin, which received
// funding from the Centre national d'études spatiales (CNES).

import qe from './query-engine-alasql'
import registerPromiseWorker from 'promise-worker/register'

registerPromiseWorker((message) => {
  if (message.type === 'initDB') {
    return qe.initDB(message.fieldsList)
  } else if (message.type === 'loadGeojson') {
    return qe.loadGeojson(message.url)
  } else if (message.type === 'query') {
    return qe.query(message.q)
  }
})
